python/lsst/pipe/analysis/fakesAnalysis.py

```diff
@@ -283,58 +292,96 @@ def fakesAreaDepth(inputFakesMatched, processedFakesMatched, plotInfoDict, areaD
+            magsToLimit[i] = interpSpline(numSigmas)
 
-    sigmas = ["10", "100"]
     areas = np.array(areas)/(3600.0**2)
-    for (i, depths) in enumerate([mag10s, mag100s]):
-        bins = np.linspace(min(depths), max(depths), 101)
-        areasOut = np.zeros(100)
-        n = 0
-        magLimHalfArea = None
-        magLim25 = None
-        magLim75 = None
-        while n < len(bins)-1:
-            ids = np.where((depths >= bins[n]) & (depths < bins[n + 1]))[0]
-            areasOut[n] += np.sum(areas[ids])
-            if np.sum(areasOut) > np.sum(areas) / 2.0 and magLimHalfArea is None:
-                magLimHalfArea = (bins[n] + bins[n + 1]) / 2.0
-            if np.sum(areasOut) > np.sum(areas) * 0.25 and magLim25 is None:
-                magLim25 = (bins[n] + bins[n + 1]) / 2.0
-            if np.sum(areasOut) > np.sum(areas) * 0.75 and magLim75 is None:
-                magLim75 = (bins[n] + bins[n + 1]) / 2.0
-            n += 1
-
-        cumAreas = np.zeros(100)
-        n = 0
-        for area in areasOut[::-1]:
-            cumAreas[n] = area + cumAreas[n-1]
-            n += 1
-
-        plt.plot(bins[::-1][1:], cumAreas)
-        plt.xlabel("Magnitude Limit (" + sigmas[i] + " sigma)", fontsize=15)
-        plt.ylabel("Area / deg^2", fontsize=15)
-        plt.title('Total Area to Given Depth \n (Recovered fake stars with no match within 2")')
-        labelHA = "Mag. Lim. for 50% of the area: {:0.2f}".format(magLimHalfArea)
-        plt.axvline(magLimHalfArea, label=labelHA, color="k", ls=":")
-        label25 = "Mag. Lim. for 25% of the area: {:0.2f}".format(magLim25)
-        plt.axvline(magLim25, label=label25, color="k", ls="--")
-        label75 = "Mag. Lim. for 75% of the area: {:0.2f}".format(magLim75)
-        plt.axvline(magLim75, label=label75, color="k", ls="--")
-        plt.legend(loc="best")
-
-        fig = plt.gcf()
-        fig = addProvenanceInfo(fig, plotInfoDict)
-
-        repoInfo.dataId["description"] = "areaDepth" + sigmas[i]
-        repoInfo.butler.put(fig, "plotFakes", repoInfo.dataId)
-        plt.close()
-
-    medMag10 = np.median(mag10s)
-    medMag100 = np.median(mag100s)
-
-    return medMag10, medMag100
+    bins = np.linspace(min(magsToLimit), max(magsToLimit), 101)
+    areasOut = np.zeros(100)
+    n = 0
+    magLimHalfArea = None
+    magLim25 = None
+    magLim75 = None
+    while n < len(bins)-1:
+        ids = np.where((magsToLimit >= bins[n]) & (magsToLimit < bins[n + 1]))[0]
+        areasOut[n] += np.sum(areas[ids])
+        if np.sum(areasOut) > np.sum(areas) / 2.0 and magLimHalfArea is None:
+            magLimHalfArea = (bins[n] + bins[n + 1]) / 2.0
+        if np.sum(areasOut) > np.sum(areas) * 0.25 and magLim25 is None:
+            magLim25 = (bins[n] + bins[n + 1]) / 2.0
+        if np.sum(areasOut) > np.sum(areas) * 0.75 and magLim75 is None:
+            magLim75 = (bins[n] + bins[n + 1]) / 2.0
+        n += 1
+
+    cumAreas = np.zeros(100)
```

Comment:
Does this block do the same thing as `cumAreas = numpy.cumsum(areasOut[::-1])`?

Reply:
I think it doesn't quite. It might be able to be prodded into using that but it is only ever used on a very small (100) list so I didn't put thought into optimising it.